units/generate_tasks.go

```diff
@@ -259,6 +265,27 @@ func (j *generateTasksJob) Run(ctx context.Context) {
 		return
 	}
 
+	defer func() {
+		pErr := recovery.HandlePanicWithError(recover(), nil, fmt.Sprintf("panic in %s job", generateTasksJobName))
+		if err == nil && pErr == nil {
+			return
+		}
+		grip.Error(message.Fields{
+			"message":   "generate.tasks error, may retry",
+			"operation": generateTasksJobName,
+			"task":      j.TaskID,
+			"execution": j.Execution,
+			"attempt":   j.Attempt,
+			"job":       j.ID(),
+			"version":   t.Version,
+			"error":     err,
+			"panic":     pErr,
+		})
+		if j.Attempt < generateTasksJobMaxAttempts-1 {
```

Comment:
Should we log when there are no attempts remaining?

Reply:
added an extra line